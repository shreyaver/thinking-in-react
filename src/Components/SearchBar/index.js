import React from 'react';
import PropTypes from 'prop-types';

// handleFilterTextChange = (newFilterText) => {
//     this.props.changeFilterTextState(newFilterText.target.value);
// }
// handleInStockOnlyChange = (newInStockOnly) => {
//     this.props.changeInStockOnlyState(newInStockOnly.target.checked);
// }
const SearchBar = props => (
  <form>
    <input type="text" className="FilterSearchBox" placeholder="Search..." value={props.filterText} onChange={newFilterText => props.changeFilterTextState(newFilterText.target.value)} />
    <br />
    <input type="checkbox" className="InStockCheckBox" checked={props.inStockOnly} onChange={newInStockOnly => props.changeInStockOnlyState(newInStockOnly.target.checked)} />
Only show products in stock
  </form>
);

SearchBar.propTypes = {
  filterText: PropTypes.string,
  inStockOnly: PropTypes.bool,
};
SearchBar.defaultProps = {
  filterText: '',
  inStockOnly: false,
};
export default SearchBar;
