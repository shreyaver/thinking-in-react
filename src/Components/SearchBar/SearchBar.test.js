import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SearchBar from '.';

describe('SearchBar', () => {
  it('renders without crashing', () => {
    const searchBar = renderer.create(<SearchBar />).toJSON();
    expect(searchBar).toMatchSnapshot();
  });
  it('sets value of checkbox according to passed value', () => {
    const wrapper = shallow(<SearchBar filterText="" inStockOnly />);
    expect(wrapper.find('.InStockCheckBox').props().checked).toEqual(true);
  });
  it('sets value of searchbox according to passed value', () => {
    const wrapper = shallow(<SearchBar filterText="abc" inStockOnly={false} />);
    expect(wrapper.find('.FilterSearchBox').props().value).toEqual('abc');
  });
  it('calls function when checkbox is toggled', () => {
    const changeInStockOnlyStateMock = jest.fn(x => x);
    const wrapper = shallow(<SearchBar filterText="" inStockOnly={false} changeInStockOnlyState={changeInStockOnlyStateMock} />);
    wrapper.find('.InStockCheckBox').simulate('change', { target: { checked: true } });
    expect(changeInStockOnlyStateMock).toHaveBeenCalled();
  });
  it('calls function when text is entered in searchbox', () => {
    const changeFilterTextStateMock = jest.fn(x => x);
    const wrapper = shallow(<SearchBar filterText="" inStockOnly={false} changeFilterTextState={changeFilterTextStateMock} />);
    wrapper.find('.FilterSearchBox').simulate('change', { target: { value: 'ball' } });
    expect(changeFilterTextStateMock).toHaveBeenCalled();
  });
  afterAll(() => {
    jest.restoreAllMocks();
  });
});
