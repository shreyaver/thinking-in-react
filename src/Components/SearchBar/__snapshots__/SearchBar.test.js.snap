// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders without crashing 1`] = `
<form>
  <input
    className="FilterSearchBox"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
    value=""
  />
  <br />
  <input
    checked={false}
    className="InStockCheckBox"
    onChange={[Function]}
    type="checkbox"
  />
  Only show products in stock
</form>
`;
